modules/effects/src/concat_latest_from.ts

```diff
@@ -0,0 +1,38 @@
+import { Observable, of, OperatorFunction } from 'rxjs';
+import { concatMap, withLatestFrom } from 'rxjs/operators';
+
+type TypeOfObservable<T> = T extends Observable<infer U> ? U : never;
+
+/**
+ * 'concatLatestFrom' combines the source value
+ * and the last available value from a lazily evaluated Observable
+ * in a new array
+ */
+export function concatLatestFrom<T extends Observable<unknown>[], V>(
+  observablesFactory: (value: V) => [...T]
+): OperatorFunction<V, [V, ...{ [i in keyof T]: TypeOfObservable<T[i]> }]>;
+export function concatLatestFrom<T extends Observable<unknown>, V>(
+  observableFactory: (value: V) => T
+): OperatorFunction<V, [V, TypeOfObservable<T>]>;
+export function concatLatestFrom<
+  T extends Observable<unknown>[] | Observable<unknown>,
+  V
+>(
+  observablesFactory: (value: V) => T extends Observable<unknown>[] ? [...T] : T
+) {
+  return concatMap((value) => {
```

Comment:
Btw, when I write custom operators, I typically do 
```ts
function myOperator(foo: string): (source: Observable<A>) => Observable<b> {
   return source.pipe(
   .... // here goes the implementation
   )
}
```

I don't quite remember why that's the way to do it. E.g. @cartant also uses it in his article: https://ncjamieson.com/managing-operator-state/

Might be a good idea to do it this way anyway. (in the meantime, I'll try to find out why it's done so).

e.g. here's what I also arrived at for this operator: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgeQEYGcCmUBuBDVAG0wBoUNt8jMBJAOzAFcYy0tdMATANT0MczIAZmQhC4AXzhCoEEHADkUAB4ArdAoDcAKFCRYiOAGMIdI3hgBZPGDIB3YDAAWAGQuZ0MAGKz5UmXKKKuoA9BBg2BbQGjraISFwAIJQUHgAnnAQONiEEHiccMDocHSYXFzSwFCeAHTamMr68EKMZjDApsam5jBuMB7evgA8ACpwDf10nMVslATEQ60A1nQQdnQAfADaALpk3BsAFNpwmRS48x5eeEYw0GkAXHCH+PyYT9wAlHAAvBtwWxqQJGOzg8Tg6DAUEw+WAdAA5hCIHA7JhjHg6HB4Zh4M40TBGGBiJkoJxsJlxJhiCBMHQYOhtJ8nsdTugIIwoEZ3uR2FQFgdGb9-rMLtQhltuGQgTUkFsANKFTFLTBpMRwEFPEXZHh8ATCUbynb-CRGnQNJrSVq3DqYkxmCx9AY+OSjcbKSbTHlzMXLVbrDb7I4nM68y7oa63e7M14CD7fP7qxnMtkcrma858zBDA7x4UZy7iyVejg6t76kYbU1xABU1ZO1cUdp6js8zpACi6IFQcI8cDxSNTaJjmHrcAxBX7hDwnjH+GAU+ocGH0l8Y7gU4AXvOMpgY+4ClrLqO4WvSnYxyl0vWQvVGtBmlb2p0mw73K3hsGxhNaZ7Dz66CsaybHAAA+xaZosAF+psuwkMGkrBgASr8ALBqcCGnKc0qHF+7o-sUySpGkkGAf6aGYQA-IY8qKnAyqquIGrgVwvBlkIBpykakjkacTxbCCnzBjs2hBqcED5tQ4Y3HcUCPM8w5xkKiZMs8Kactyf78hsubgQWiH-AgwbQgSUCYocalprpYo5kK5EWZgNRgMAEQsphpwvlYNiHC8uqYDphluW5dozuJoaSShoXesQUmRrJPlvJ8OiBZhwXwJForRYk6CEekKE5WkNRFPlhzpZm6CCclblUSVEnRWOMy1bQDDMCR0HbDsFWVbxAKlWGOxJZVxkcpiYjxQInyOc5mCuV1DjOC2gxyIc0q9ZJWX5Z1yXfNOVkLPpA1uRIm1wIl2gSEAA

Reply:
Since writing this PR, I have spent more time on the RxJs docs site and [found that they give the same recommendation](https://rxjs.dev/guide/operators). So I'll swap it over.
Thanks for the playground link 👍.

Reply:
Revising my statement- RxJs now provides the `pipe` function to independently and more quickly create custom operators. No need to spend the extra characters referring to a source and writing details that expose a deeper understanding than necessary of the nature of observables.